lib/_http_outgoing.js

```
@@ -650,7 +650,7 @@ function matchHeader(self, state, field, value) {
       break;
     case 'content-length':
       state.contLen = true;
-      self._contentLength = value;
+      self._contentLength = ~~value;
```

As @aduh95 noted, this has a few weird edge cases, and also does things like rounding numbers automatically (which is weird - `setHeader('content-length', '1.5')` should really just fail somewhere rather than rounding) and returning `0` instead of `NaN` (`~~"hello" === 0` - not good!)

If we're looking for a quick coercion `+` seems like the best option to me - no funny business with rounding/NaN, and `+((2**31).toString()) === (2**31)`

Although it's relevant to other cases, I don't think `ERR_INVALID_TYPE` is relevant here, as setting headers as string is really the _expected_ behaviour - HTTP headers are fundamentally strings, not numbers, and there's lots of cases where that will be apparent. This is just a special case where we need to process the value internally and use it numerically as well as just sending it.